pkg/p2p/libp2p/internal/handshake/handshake_test.go

```diff
@@ -164,37 +224,80 @@ func TestHandshake(t *testing.T) {
 			t.Fatal("res should be nil")
 		}
 
-		if err != ErrNetworkIDIncompatible {
-			t.Fatalf("expected %s, got %s", ErrNetworkIDIncompatible, err)
+		if err == nil || err.Error() != "invalid ack received" {
```

Comment:
Also here, it is better to check for the concrete error, then for the error message string.